src/rpc-browser-ws.ts

```diff
@@ -2,15 +2,18 @@
 // see: https://github.com/microsoft/TypeScript/issues/16577#issuecomment-343610106
 
 import { RpcCommon, IPromiseCallbacks } from "./rpc-common.js";
+import { IChildLogger } from "@vscode-logging/types";
+import { noopLogger } from "./noop-logger";
 
 export class RpcBrowserWebSockets extends RpcCommon {
   ws: WebSocket;
 
-  constructor(ws: WebSocket) {
-    super();
+  constructor(ws: WebSocket, logger: IChildLogger = noopLogger) {
+    super(logger);
     this.ws = ws;
     this.ws.addEventListener("message", (event) => {
       const message: any = JSON.parse(event.data as string);
+      this.logger.debug(`Received event: ${message.command} id: ${message.id} method: ${message.method} params: ${message.params}`);
```

Comment:
```suggestion
      this.logger.debug(`Received event: ${_.get(message, 'command', '')} id: ${_.get(message, 'id', '')} method: ${_.get(message, 'method', '')} params: ${_.get(message, 'params', '')}`);
```

Reply:
JSON.parse will throw  SyntaxError exception if it cannot parse data as valid json, there is no way message is undefined here

Reply:
you are right. but message is declared as any, therefore it can be any valid object but not obligated to have the attributes you expected : "command", "params" and etc.